server/project.go

```diff
@@ -269,6 +269,19 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("VCS not found with ID: %d", repositoryCreate.VCSID))
 		}
 
+		// Make sure the branch exists in the repo.
+		if _, err := vcsPlugin.Get(vcs.Type, vcsPlugin.ProviderConfig{}).GetBranch(ctx,
```

Comment:
We're supporting the branch with a wildcard. Is there a way to filter this? Is there a way to share code with below?

Reply:
I've updated the code to skip checks for branch names with wildcards. 
Should we also apply checks to branch names with wildcards? Like for the given branch name `main*`, we check if the repo has any branch that starts with `main`. If none, we throw an error?

Reply:
That's probably too much work. This is good for now.